src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java

```diff
@@ -182,7 +186,13 @@ public Object newInstance(Object ctrParamObj)
                     }
                 } else  if (type.equals("KeyAgreement")) {
                     if (algo.equals("ECDH")) {
-                        if (useNativeEC) {
+                        boolean isNativeCryptoLoadAllowed = true;
+/*[IF CRIU_SUPPORT]*/
+                        if (InternalCRIUSupport.isCheckpointAllowed()) {
+                            isNativeCryptoLoadAllowed = false;
+                        }
+/*[ENDIF] CRIU_SUPPORT*/
+                        if (useNativeEC && isNativeCryptoLoadAllowed && NativeCrypto.isLoaded()) {
```

Comment:
Suggest we don't need the extra local variable:
```java
                        if (useNativeEC
/*[IF CRIU_SUPPORT]*/
                            && !InternalCRIUSupport.isCheckpointAllowed()
/*[ENDIF] CRIU_SUPPORT */
                            && NativeCrypto.isLoaded()
                        ) {
```

Reply:
Updated